### ISSUE TITLE
Still getting infinite axis errors when plotting cox.zph for each covariate

### ISSUE THREAD

**Issue by dchiu911** [therneau/survival#90]
I just installed survival 3.1-8, and I'm still getting errors with `plot(cox.zph(fit)`, where `fit` has a `strata` term. I also want a plot for each covariate.

## Previous Output (2.44.1-1)
``` r
library(survival)
test1 <- list(time = c(4, 3, 1, 1, 2, 2, 3),
              status = c(1, 1, 1, 0, 1, 1, 0),
              x = c(0, 2, 1, 1, 1, 0, 0),
              sex = c(0, 0, 0, 0, 1, 1, 1))
test1$x <- factor(test1$x)
fit <- coxph(Surv(time, status) ~ x + strata(sex), test1)
temp <- cox.zph(fit)
print(temp)
#>          rho   chisq     p
#> x1     0.018 0.00165 0.968
#> x2     0.560 0.56180 0.454
#> GLOBAL    NA 0.61129 0.737
plot(temp)
```

![](https://i.imgur.com/f9YBFte.png)![](https://i.imgur.com/3cZoyMh.png)

<sup>Created on 2019-12-03 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

## Current Output (3.1-8)
``` r
library(survival)
test1 <- list(time = c(4, 3, 1, 1, 2, 2, 3),
              status = c(1, 1, 1, 0, 1, 1, 0),
              x = c(0, 2, 1, 1, 1, 0, 0),
              sex = c(0, 0, 0, 0, 1, 1, 1))
test1$x <- factor(test1$x)
fit <- coxph(Surv(time, status) ~ x + strata(sex), test1)
temp <- cox.zph(fit, terms = FALSE)
print(temp)
#>        chisq df    p
#> x1     0.903  1 0.34
#> x2     0.710  1 0.40
#> GLOBAL 0.976  2 0.61
plot(temp)
#> Warning in min(x): no non-missing arguments to min; returning Inf
#> Warning in max(x): no non-missing arguments to max; returning -Inf
```

![](https://i.imgur.com/q37l3Dj.png)

    #> Error in plot.window(...): need finite 'ylim' values

![](https://i.imgur.com/QRGbm7i.png)

<sup>Created on 2019-12-03 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by therneau** [comment]
I spoke too soon.   I have fixed an error in cox.zph (fixed in 1.3-8), but this is a different one.
I’ll look into it.


Sent from my iPad

On Dec 3, 2019, at 12:32 PM, Derek Chiu <notifications@github.com<mailto:notifications@github.com>> wrote:


I just installed survival 3.1-8, and I'm still getting errors with plot(cox.zph(fit), where fit has a strata term. I also want a plot for each covariate.

library(survival)
test1 <- list(time = c(4, 3, 1, 1, 2, 2, 3),
              status = c(1, 1, 1, 0, 1, 1, 0),
              x = c(0, 2, 1, 1, 1, 0, 0),
              sex = c(0, 0, 0, 0, 1, 1, 1))
test1$x <- factor(test1$x)
fit <- coxph(Surv(time, status) ~ x + strata(sex), test1)
temp <- cox.zph(fit, terms = FALSE)
plot(temp)
#> Warning in min(x): no non-missing arguments to min; returning Inf
#> Warning in max(x): no non-missing arguments to max; returning -Inf

[https://camo.githubusercontent.com/fd28fda91b46cca23c0bba8d72ebeaf1d8e3a39c/68747470733a2f2f692e696d6775722e636f6d2f57556a727464702e706e67]<https://camo.githubusercontent.com/fd28fda91b46cca23c0bba8d72ebeaf1d8e3a39c/68747470733a2f2f692e696d6775722e636f6d2f57556a727464702e706e67>

#> Error in plot.window(...): need finite 'ylim' values


[https://camo.githubusercontent.com/8ca6b39c0e1b4f0ca238257a88c1227e467a84a2/68747470733a2f2f692e696d6775722e636f6d2f4a71354b53534f2e706e67]<https://camo.githubusercontent.com/8ca6b39c0e1b4f0ca238257a88c1227e467a84a2/68747470733a2f2f692e696d6775722e636f6d2f4a71354b53534f2e706e67>

Created on 2019-12-03 by the reprex package<https://reprex.tidyverse.org> (v0.3.0)

**Comment by therneau** [comment]
This should now be repaired.
